ui/utils.js

```diff
@@ -232,6 +232,17 @@ function suggest_element_until_click(selector, state_to_set) {
     });
 }
 
+function format_asset_title_for_ui(asset) {
```

Comment:
I have some changes to suggest for this function.

1. In my system when I installed the npm package, the icons were all lower-case. So this fails to find them.
2. This will show an ugly image not found icon for any asset that has no corresponding image. I suggest using the [generic currency sign](https://en.wikipedia.org/wiki/Currency_sign_(typography)) ¤ instead.

Here is a suggested patch on top of your code to achieve this:

```diff
diff --git a/ui/utils.js b/ui/utils.js                                                                                                                                                                                                                                          
index bc57b7e..29a5e3d 100644
--- a/ui/utils.js
+++ b/ui/utils.js
@@ -233,14 +233,24 @@ function suggest_element_until_click(selector, state_to_set) {
 }
 
 function format_asset_title_for_ui(asset) {
-  var asset_call_letters = '';
-  if(asset == 'IOTA') {
-    asset_call_letters = 'MIOTA';
-  } else {
-    asset_call_letters = asset;
-  }
-  var str = '<img src="../node_modules/cryptocurrency-icons/svg/color/'+asset_call_letters+'.svg" />'+' '+asset;
-  return str;
+    var asset_call_letters = '';
+    var str;
+    if(asset == 'IOTA') {
+        asset_call_letters = 'MIOTA';
+    } else {
+        asset_call_letters = asset;
+    }
+    resolve = require('path').resolve;
+    let path = 'node_modules/cryptocurrency-icons/svg/color/' + asset_call_letters.toLowerCase() + '.svg';
+    let resolved_path = resolve(path);
+    // TODO: Better import all icon names into a map and check the map instead
+    // of doing a syscall for each file existing in the user's system
+    if (fs.existsSync(resolved_path)) {
+        str = '<img src="../' + path + '" />' + ' ' + asset;
+    } else {
+        str = ' ' + asset;
+    }
+    return str;
 }
 
 module.exports = function() {
```

Note I also changed the style and fixed indentation to be 4 spaces.

Reply:
Come to think of it, I think mine were installed lower-case also, so I'm surprised I hadn't ran into that bug sooner. I would have implemented the toLower function call at that time.
I absolutely love your suggestion to use the generic currency sign instead of that ugly image that showed from my screenshot. Thank you so much for this patch, it is very useful and I'm so glad for your assistance.

I noticed the style changes also and have made the mental note moving forward with this project. I was unaware to the standard previously.

I will make the changes this evening.